File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGauges.Win.Gauges.Linear;
using System.Diagnostics;
using DevExpress.XtraGauges.Core.Base;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }


        Size gaugeScaleProportion = new Size(523, 115);
        float markerSizeProportion = 0.2f;
        
        private void Form1_Load(object sender, EventArgs e)
        {
            UpdateGaugeSize(linearGauge1, gaugeControl1.Size, gaugeScaleProportion, markerSizeProportion); 
        }

        private void UpdateGaugeSize(LinearGauge gauge, SizeF containerSize, SizeF scaleProportion, float markerProportion)
        {
            float indentX = Math.Min(containerSize.Width / scaleProportion.Width * scaleProportion.Height * markerProportion, containerSize.Height * markerProportion);
            float indentY = containerSize.Height / 8f;
            SizeF targetArea = new SizeF(containerSize.Width - indentX * 2f, containerSize.Height - indentY * 2f);

            float proportion = Math.Max(1f, ((targetArea.Width / scaleProportion.Width) / (targetArea.Height / scaleProportion.Height)));
            gauge.Bounds = new Rectangle((int)indentX, (int)(indentY + (targetArea.Height - targetArea.Width / 2f / proportion) / 2f), (int)targetArea.Width, (int)(targetArea.Width / 2f / proportion));
            gauge.Scales[0].StartPoint = new PointF2D(62.5f, 125f * (1f + proportion));
            gauge.Scales[0].EndPoint = new PointF2D(62.5f, 125f * (1f - proportion));
        }

        private void Form1_Resize(object sender, EventArgs e)
        {
            UpdateGaugeSize(linearGauge1, gaugeControl1.Size, gaugeScaleProportion, markerSizeProportion); 
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGauges.Win.Gauges.Linear
Imports System.Diagnostics
Imports DevExpress.XtraGauges.Core.Base

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub


        Private gaugeScaleProportion As New Size(523, 115)
        Private markerSizeProportion As Single = 0.2F

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            UpdateGaugeSize(linearGauge1, gaugeControl1.Size, gaugeScaleProportion, markerSizeProportion)
        End Sub

        Private Sub UpdateGaugeSize(ByVal gauge As LinearGauge, ByVal containerSize As SizeF, ByVal scaleProportion As SizeF, ByVal markerProportion As Single)
            Dim indentX As Single = Math.Min(containerSize.Width / scaleProportion.Width * scaleProportion.Height * markerProportion, containerSize.Height * markerProportion)
            Dim indentY As Single = containerSize.Height / 8F
            Dim targetArea As New SizeF(containerSize.Width - indentX * 2F, containerSize.Height - indentY * 2F)

            Dim proportion As Single = Math.Max(1F, ((targetArea.Width \ scaleProportion.Width) \ (targetArea.Height \ scaleProportion.Height)))
            gauge.Bounds = New Rectangle(CInt((indentX)), CInt((indentY + (targetArea.Height - targetArea.Width / 2F / proportion) / 2F)), CInt(targetArea.Width), CInt((targetArea.Width / 2F / proportion)))
            gauge.Scales(0).StartPoint = New PointF2D(62.5F, 125F * (1F + proportion))
            gauge.Scales(0).EndPoint = New PointF2D(62.5F, 125F * (1F - proportion))
        End Sub

        Private Sub Form1_Resize(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Resize
            UpdateGaugeSize(linearGauge1, gaugeControl1.Size, gaugeScaleProportion, markerSizeProportion)
        End Sub
    End Class
End Namespace
